File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/examples-java/Monitor_Upgrade.java
/** Samples for MonitorOperation Upgrade. */
public final class Main {
    /*
     * x-ms-original-file: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/examples/Monitor_Upgrade.json
     */
    /**
     * Sample code: Monitor_Upgrade.
     *
     * @param manager Entry point to ElasticManager.
     */
    public static void monitorUpgrade(com.azure.resourcemanager.elastic.ElasticManager manager) {
        manager.monitorOperations().upgrade("myResourceGroup", "myMonitor", null, com.azure.core.util.Context.NONE);
    }
}

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/examples-js/Monitor_Upgrade.js
const { MicrosoftElastic } = require("@azure/arm-elastic");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Upgradable version for a monitor resource.
 *
 * @summary Upgradable version for a monitor resource.
 * x-ms-original-file: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/examples/Monitor_Upgrade.json
 */
async function monitorUpgrade() {
  const subscriptionId =
    process.env["ELASTIC_SUBSCRIPTION_ID"] || "00000000-0000-0000-0000-000000000000";
  const resourceGroupName = process.env["ELASTIC_RESOURCE_GROUP"] || "myResourceGroup";
  const monitorName = "myMonitor";
  const credential = new DefaultAzureCredential();
  const client = new MicrosoftElastic(credential, subscriptionId);
  const result = await client.monitor.beginUpgradeAndWait(resourceGroupName, monitorName);
  console.log(result);
}
